internal/provider/resource_aws_transit_gateway_attachment.go

```diff
@@ -0,0 +1,316 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/client/network_service"
+	networkmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+var tgwDefaultTimeout = time.Minute * 1
+var tgwCreateTimeout = time.Minute * 35
+var tgwDeleteTimeout = time.Minute * 35
+
+func resourceAwsTransitGatewayAttachment() *schema.Resource {
+	return &schema.Resource{
+		Description: "The AWS Transit gateway attachment resource allows you to manage a Transit gateway attachment that attaches an HVN to a Transit gateway in AWS.",
+
+		CreateContext: resourceAwsTransitGatewayAttachmentCreate,
+		ReadContext:   resourceAwsTransitGatewayAttachmentRead,
+		DeleteContext: resourceAwsTransitGatewayAttachmentDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &tgwDefaultTimeout,
+			Create:  &tgwCreateTimeout,
+			Delete:  &tgwDeleteTimeout,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"hvn_id": {
+				Description:      "The ID of the HashiCorp Virtual Network (HVN).",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"transit_gateway_attachment_id": {
+				Description:      "The ID of the Transit gateway attachment.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"transit_gateway_id": {
+				Description: "The ID of the Transit gateway in AWS.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"resource_share_arn": {
+				Description: "The Amazon Resource Name (ARN) of the Resource Share that is needed to grant HCP access to the Transit gateway in AWS.",
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
```

Comment:
How come the ARN is sensitive in this case? I usually think of AWS ARNs synonymous to unique IDs, have I been lying to myself this whole time!?

Reply:
Hmm, this came from @smaant , but I haven't actually verified. Anton, do you know if this is a sensitive value?

Reply:
in general ARNs aren't sensitive but we kind of treat them as such. The idea is that Resource Share (RS) is shared with the entire HVN's org AWS account, therefore technically any HVN in this org can use this RS to attach to TGW shared in this RS as long as they know TGW ID and RS ARN therefore we need to keep one them as a secret.